lib/DeckClass.php

```diff
@@ -88,9 +93,11 @@ public function addCard($data, $user, $board = null) {
             $card = $this->apiCall("POST", NC_SERVER . "/index.php/apps/deck/api/v1.0/boards/{$params->board}/stacks/{$params->stack}/cards", $data);
             $card->board = $params->board;
             $card->stack = $params->stack;
-    
+
+            if ($params->userId !== null) $user->userId = $params->userId;
+
             if($this->responseCode == 200) {
-                if(ASSIGN_SENDER) $this->assignUser($card, $user);
+                if(ASSIGN_SENDER || $params->userId !== null) $this->assignUser($card, $user);
```

Comment:
It looks like you are only trying to make sure userId is not exactly null on $params. I assume you just want to make sure it exists, therefore you could do it without the type check?

```suggestion
                if(ASSIGN_SENDER || $params->userId) $this->assignUser($card, $user);
```